### PR TITLE
Support ES version 7.6.1

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
*Issue #, if available:*

*Description of changes:*
Support ES Version 7.6.1

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-tools/esplugin-coverage.gradle [hunk 1]
```diff
@@ -65,30 +69,17 @@ integTestCluster {
 jacocoTestReport {
     dependsOn integTest, test
     executionData dummyTest.jacoco.destinationFile, dummyIntegTest.jacoco.destinationFile
-    sourceDirectories = sourceSets.main.allSource
-    classDirectories = sourceSets.main.output
+    // sourceDirectories.from = sourceSets.main.allSource
```
Comment (dbbaughe) [inline build-tools/esplugin-coverage.gradle]: Remove commented out part

File: build.gradle [hunk 2]
```diff
@@ -15,15 +15,14 @@
...
 buildscript {
     ext {
-        es_version = System.getProperty("es.version", "7.4.2")
-        kotlin_version = System.getProperty("kotlin.version", "1.3.31")
+        es_version = System.getProperty("es.version", "7.6.1")
+        kotlin_version = System.getProperty("kotlin.version", "1.3.61")
     }
 
     repositories {
+        mavenLocal()
```
Comment (dbbaughe) [inline build.gradle]: Why was this moved above other repositories?

File: build.gradle [hunk 3]
```diff
@@ -71,14 +71,29 @@ detekt {
     buildUponDefaultConfig = true
 }
 
+ext {
```
Comment (dbbaughe) [inline build.gradle]: Reason for moving?

File: build.gradle [hunk 4]
```diff
@@ -125,52 +126,25 @@ test {
     systemProperty 'tests.security.manager', 'false'
 }
 
-integTestRunner {
+testClusters.integTest {
+    testDistribution = "OSS"
+    // Temporary until job-scheduler is published to Maven
```
Comment (dbbaughe) [inline build.gradle]: Should be able to remove this comment now

File: opendistro-elasticsearch-index-management.release-notes.md [hunk 5]
```diff
@@ -1,3 +1,15 @@
+## Version 1.6.0.0 (2020-03-17)
```
Comment (dbbaughe) [inline opendistro-elasticsearch-index-management.release-notes.md]: You have SQL PRs in these release notes

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt [hunk 6]
```diff
@@ -112,7 +112,7 @@ class WaitForForceMergeStep(
             val statsResponse: IndicesStatsResponse = client.admin().indices().suspendUntil { stats(statsRequest, it) }
 
             if (statsResponse.status == RestStatus.OK) {
-                return statsResponse.shards.count { it.stats.segments.count > maxNumSegments }
+                return statsResponse.shards.count { it.stats.segments!!.count > maxNumSegments }
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt]: Don't use !! in production code, using it in our tests is fine, this opens up possibility for NPEs

Comment (dbbaughe) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt]: Multiple places below too

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/NotificationActionIT.kt [hunk 7]
```diff
@@ -40,12 +40,14 @@ class NotificationActionIT : IndexStateManagementRestTestCase() {
...
         val policyID = "${testIndexName}_testPolicyName"
         val notificationIndex = "notification_index"
+        val clusterUri = System.getProperty("tests.rest.cluster").split(",")[0]
         val destination = Destination(
             type = DestinationType.CUSTOM_WEBHOOK,
             chime = null,
             slack = null,
             customWebhook = CustomWebhook(
-                url = "http://${System.getProperty("tests.rest.cluster")}/$notificationIndex/_doc",
+                // url = "http://${System.getProperty("tests.rest.cluster")}/$notificationIndex/_doc",
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/NotificationActionIT.kt]: Remove commented code

File: build.gradle (hunk 2 above)
Comment (bowenlan-amzn) [reply build.gradle]: Though it helps gradle to find local publish

File: build.gradle (hunk 3 above)
Comment (bowenlan-amzn) [reply build.gradle]: ......no reason, my bad

File: opendistro-elasticsearch-index-management.release-notes.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
...
+
+### New Features
+* Adds support for Elasticsearch 7.6.1 [PR #164](https://github.com/opendistro-for-elasticsearch/index-management/pull/164)
```
Comment (dbbaughe) [inline opendistro-elasticsearch-index-management.release-notes.md]: We should provide details about all the changes for development.

File: README.md [hunk 9]
```diff
@@ -74,31 +74,29 @@ This project currently uses the Notification subproject from the [Alerting plugi
 4. `./gradlew integTest -Dtests.class=*RestChangePolicyActionIT` runs a single integ class
 5.  `./gradlew integTest -Dtests.class=*RestChangePolicyActionIT -Dtests.method="test missing index"` runs a single integ test method (remember to quote the test method name if it contains spaces)
 
-When launching a cluster using one of the above commands, logs are placed in `build/cluster/run node0/elasticsearch-<version>/logs`. Though the logs are teed to the console, in practices it's best to check the actual log file.
+When launching a cluster using one of the above commands, logs are placed in `build/testclusters/integTest-0/logs`. Though the logs are teed to the console, in practices it's best to check the actual log file.
 
 ### Debugging
```
Comment (dbbaughe) [inline README.md]: Doesn't look like we addressed all the debugging issues?
* Can I debug the elasticsearch server when running ./gradlew run
* Can I debug the integration test when running ./gradlew integTest
* Can I debug the elasticsearch server when running ./gradlew integTest

Looks like you are missing the last one.

File: README.md [hunk 10]
```diff
@@ -74,31 +74,29 @@ This project currently uses the Notification subproject from the [Alerting plugi
...
-Sometimes it's useful to attach a debugger to either the Elasticsearch cluster or the integ tests to see what's going on. When running unit tests, hit **Debug** from the IDE's gutter to debug the tests.  To debug code running in an actual server, run:
+When running unit tests, hit **Debug** from the IDE's gutter to debug the tests.  
 
-```
-./gradlew integTest --debug-jvm # to start a cluster and run integ tests
-```
+Sometimes it's useful to attach a debugger to either the Elasticsearch cluster or the integ tests to see what's going on. 
 
-OR
+To debug code running in an integ test (which exercises the server from a separate JVM), run:
```
Comment (dbbaughe) [inline README.md]: Confused by these changes, looks like you just moved the `./gradlew -Dtest.debug integTest` section above the `./gradlew run --debug-jvm` section. Why?

File: README.md [hunk 11]
```diff
@@ -74,31 +74,29 @@ This project currently uses the Notification subproject from the [Alerting plugi
...
-When launching a cluster using one of the above commands, logs are placed in `build/cluster/run node0/elasticsearch-<version>/logs`. Though the logs are teed to the console, in practices it's best to check the actual log file.
+When launching a cluster using one of the above commands, logs are placed in `build/testclusters/integTest-0/logs`. Though the logs are teed to the console, in practices it's best to check the actual log file.
 
 ### Debugging
 
-Sometimes it's useful to attach a debugger to either the Elasticsearch cluster or the integ tests to see what's going on. When running unit tests, hit **Debug** from the IDE's gutter to debug the tests.  To debug code running in an actual server, run:
+When running unit tests, hit **Debug** from the IDE's gutter to debug the tests.  
 
-```
-./gradlew integTest --debug-jvm # to start a cluster and run integ tests
```
Comment (dbbaughe) [inline README.md]: This needs to not be removed. It needs to be fixed instead. Please refer to: https://github.com/opendistro-for-elasticsearch/alerting/commit/f3a94cd5a871cda7e01a5841746442fc1de9149a

Comment (bowenlan-amzn) [reply README.md]: Another reference: https://github.com/opendistro-for-elasticsearch/k-NN/pull/78

File: README.md (hunk 10 above)
Comment (bowenlan-amzn) [reply README.md]: change back, sorry

File: README.md (hunk 9 above)
Comment (bowenlan-amzn) [reply README.md]: Added and Tested

File: opendistro-elasticsearch-index-management.release-notes.md (hunk 8 above)
Comment (bowenlan-amzn) [reply opendistro-elasticsearch-index-management.release-notes.md]: Add some, think these are good and enough.

File: opendistro-elasticsearch-index-management.release-notes.md [hunk 12]
```diff
@@ -1,3 +1,10 @@
...
+* Add github action workflow for CI/CD [PR #161](https://github.com/opendistro-for-elasticsearch/index-management/pull/161) and release [PR #163](https://github.com/opendistro-for-elasticsearch/index-management/pull/163)
```
Comment (dbbaughe) [inline opendistro-elasticsearch-index-management.release-notes.md]: These aren't added yet, they are still open PRs. Only add things that have been merged already (and the this open PR that you know will be merged).
